web_monitoring/html_diff_render.py

```diff
@@ -187,8 +187,74 @@
 MAX_SPACERS = 2500
 
 
+class WaybackUrlComparator:
+    matcher = re.compile(r'web/\d{14}(im_)?/(https?://)?(www.)?')
```

Comment:
> Finally, in the ending part, is there a case where it would not be an absolute URL? i.e. is (https?://)? really optional? If not, we should probably also remove the (www.)? after it, since that isn’t really doing anything either. Should this also be restricted to http, either? Couldn’t some of these URLs be ftp://, etc?

We could have http://www.anexample......
                        http://anexample......
                        https://www.anexample......
                        https://anexample......
So I believe the regex is correct.
I haven't run into any `ftp://` 's etc so far. @vbanos is this a possibility?

Reply:
> Based on that, it looks it should probably be (id_|im_|js_|cs_)?.
There can't be _id in a snapshot but the rest should be added.

Reply:
So, based on that, it looks like the protocol, at least, is not optional, so you should not have `(https?://)?`, you should have `https?://`.

Reply:
Sorry, I omitted the examples with the protocol. Again it might or might not be there.